pkg/csi/service/node.go

```diff
@@ -77,3 +285,77 @@ func (s *service) NodeGetInfo(
 
 	return nil, nil
 }
+
+// Device is a struct for holding details about a block device
+type Device struct {
+	FullPath string
+	Name     string
+	RealDev  string
+}
+
+// getDevice returns a Device struct with info about the given device, or
+// an error if it doesn't exist or is not a block device
+func getDevice(path string) (*Device, error) {
+
+	fi, err := os.Lstat(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// eval any symlinks and make sure it points to a device
+	d, err := filepath.EvalSymlinks(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO does EvalSymlinks throw error if link is to non-
+	// existent file? assuming so by masking error below
```

Comment:
Might as well just return the error.  Possible (though very unlikely) the path could be removed in between EvalSymlinks and Stat.